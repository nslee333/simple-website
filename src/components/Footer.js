import React from "react";
import github_logo from "../../public/images/github_logo_white.svg";
import Image from "next/image";

export default function Footer() {
  return (
    <>
      <div className="m-5 flex justify-center p-10">
        <a href="https://github.com/BendJS/simple-website" target="_blank">
          <div className="text-white">
            <div>Contribute to this website</div>
            <div className="flex justify-center p-2">
              <Image src={github_logo} alt="GitHub Logo" className="w-[5rem]" />
            </div>
          </div>
        </a>
      </div>
    </>
  );
}
