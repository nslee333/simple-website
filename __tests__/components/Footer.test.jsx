import { render, screen } from "@testing-library/react";
import Footer from "../../src/components/Footer";
import { toHaveAttribute } from "@testing-library/jest-dom";
import renderer from "react-test-renderer";

test("Footer renders without crashing", () => {
  render(<Footer />);
});

test("Footer renders correctly according to snapshot.", () => {
  const tree = renderer.create(<Footer />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Meetup logo has link to Meetup page.", () => {
  render(<Footer />);

  expect(
    screen.getByRole("link", { name: "Contribute to this website GitHub Logo" })
  ).toHaveAttribute("href", "https://github.com/BendJS/simple-website");
});
