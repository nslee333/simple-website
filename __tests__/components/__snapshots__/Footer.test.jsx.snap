// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly according to snapshot. 1`] = `
<div
  className="m-5 flex justify-center p-10"
>
  <a
    href="https://github.com/BendJS/simple-website"
    target="_blank"
  >
    <div
      className="text-white"
    >
      <div>
        Contribute to this website
      </div>
      <div
        className="flex justify-center p-2"
      >
        <img
          alt="GitHub Logo"
          className="w-[5rem]"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={40}
        />
      </div>
    </div>
  </a>
</div>
`;
